pkg/apis/authentication/validation/validation.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/authentication"
+)
+
+func ValidateTokenRequest(tr *authentication.TokenRequest) field.ErrorList {
```

Comment:
This should probably return a single error.
The conditions you check below are mutually exclusive anyway

Reply:
This is our validation convention https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go